pkg/virt-api/api.go

```diff
@@ -918,97 +855,60 @@ func (app *virtAPIApp) createSubresourceApiservice() error {
 	return nil
 }
 
-func (app *virtAPIApp) setupTLS(fs Filesystem) error {
-
-	app.keyFile = filepath.Join(app.certsDirectory, "/key.pem")
-	app.certFile = filepath.Join(app.certsDirectory, "/cert.pem")
-	app.signingCertFile = filepath.Join(app.certsDirectory, "/signingCert.pem")
-	app.clientCAFile = filepath.Join(app.certsDirectory, "/clientCA.crt")
-
-	// Write the certs to disk
-	err := fs.WriteFile(app.clientCAFile, app.clientCABytes, 0600)
-	if err != nil {
-		return err
-	}
-
-	if len(app.requestHeaderClientCABytes) != 0 {
-		f, err := fs.OpenFile(app.clientCAFile, os.O_APPEND|os.O_WRONLY, 0600)
-		if err != nil {
-			return err
-		}
-		defer f.Close()
-
-		_, err = f.Write(app.requestHeaderClientCABytes)
-		if err != nil {
-			return err
-		}
-	}
-
-	err = fs.WriteFile(app.keyFile, app.keyBytes, 0600)
-	if err != nil {
-		return err
-	}
-	err = fs.WriteFile(app.certFile, app.certBytes, 0600)
-	if err != nil {
-		return err
-	}
-	err = fs.WriteFile(app.signingCertFile, app.signingCertBytes, 0600)
-	if err != nil {
-		return err
-	}
-
-	// create the client CA pool.
-	// This ensures we're talking to the k8s api server
-	pool, err := cert.NewPool(app.clientCAFile)
-	if err != nil {
-		return err
-	}
-
-	app.tlsConfig = &tls.Config{
-		ClientCAs: pool,
-		// A VerifyClientCertIfGiven request means we're not guaranteed
-		// a client has been authenticated unless they provide a peer
-		// cert.
-		//
-		// Make sure to verify in subresource endpoint that peer cert
-		// was provided before processing request. If the peer cert is
-		// given on the connection, then we can be guaranteed that it
-		// was signed by the client CA in our pool.
-		//
-		// There is another ClientAuth type called 'RequireAndVerifyClientCert'
-		// We can't use this type here because during the aggregated api status
-		// check it attempts to hit '/' on our api endpoint to verify an http
-		// response is given. That status request won't send a peer cert regardless
-		// if the TLS handshake requests it. As a result, the TLS handshake fails
-		// and our aggregated endpoint never becomes available.
+func newAggregationTLSConfig(certPool *x509.CertPool, manager certificate.Manager) *tls.Config {
```

Comment:
Here I pass in the ca from the apiserver configmap for aggregation, so the same CA like the code did before. Still I do something probably obvious wrong when wiring it up, because the apiserver cert seems to be not recogniced. I get "signed by unknown CA".

Reply:
ok have it. I set the wrong CA.